presto-main/src/main/java/io/prestosql/server/security/CertificateAuthenticator.java

```diff
@@ -47,15 +51,20 @@ public Identity authenticate(HttpServletRequest request)
         if ((certs == null) || (certs.length == 0)) {
             throw new AuthenticationException(null);
         }
-        X500Principal principal = certs[0].getSubjectX500Principal();
+
         try {
-            String authenticatedUser = userExtraction.extractUser(((Principal) principal).toString());
+            Principal principal = authenticatorManager.getAuthenticator().authenticate(certs);
+            String authenticatedUser = userExtraction.extractUser((principal).toString());
             return Identity.forUser(authenticatedUser)
                     .withPrincipal(principal)
                     .build();
         }
         catch (UserExtractionException e) {
             throw new AuthenticationException(e.getMessage());
         }
+        catch (RuntimeException e) {
+            log.debug("CertificateAuthenticator plugin hasn't been loaded yet...");
```

Comment:
Change this to
```java
throw new RuntimeException("Authentication error", e);
```
I assume the "not loaded yet" is due to accepting requests while the server is initializing. This is an issue with how lifecycle works, but we don't need to solve it now. Instead, we can handle it the same way as `PasswordAuthenticator`.